runtime/jvmti/jvmtiStartup.c

```diff
@@ -1110,6 +1279,88 @@ isAgentLibraryLoaded(J9JavaVM *vm, const char *library, BOOLEAN decorate)
 			j9mem_free_memory(agentPath);
 		}
 		Trc_JVMTI_isAgentLibraryLoaded_result(library, agentPath, result);
+#elif defined(AIXPPC) /* defined(LINUX) || defined(OSX) || defined(WIN32) */
+		/* The incoming library & decorate could be in the following cases:
+		 * - JvmtiAgent1 & TRUE
+		 * - libJvmtiAgent1.so & FALSE
+		 * - /path/to/libJvmtiAgent1.so & FALSE
+		 */
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		/* retrieve a list of all object files loaded for the current process */
+		struct ld_info *objectFiles = loadqueryGetObjectFiles(vm);
+		if (NULL != objectFiles) {
+			char *libNameSo = NULL;
+			char *libNameAr = NULL;
+			char *libpath = NULL;
+			UDATA libpathSize = 0;
+			char *envLibPath = NULL;
+			char *fileName = NULL;
+			BOOLEAN isLibraryPath = FALSE;
+			struct ld_info *cursor = NULL;
+
+			if (decorate) {
+				/* the library is a platform independent name */
+				UDATA libraryLen = strlen(library);
```

Comment:
This line should appear earlier so `libraryLen` can be used on line 1247.

Reply:
Moved